tests/v2/test_peer_store_extended.nim

```diff
@@ -0,0 +1,268 @@
+      storedInfoPeer6.protos.len == 0
+      storedInfoPeer6.agent == ""
+      storedInfoPeer6.protoVersion == ""
+      storedInfoPeer6.connection == NotConnected
+      storedInfoPeer6.disconnectTime == 0
+      storedInfoPeer6.origin == Unknown
+
+  test "peers() returns all StoredInfo of the PeerStore":
+    # When
+    let allPeers = peerStore.peers()
+
+    # Then
+    check:
+      allPeers.len == 5
+      allPeers.anyIt(it.peerId == p1)
+      allPeers.anyIt(it.peerId == p2)
+      allPeers.anyIt(it.peerId == p3)
+      allPeers.anyIt(it.peerId == p4)
+      allPeers.anyIt(it.peerId == p5)
+    
+    let p3 = allPeers.filterIt(it.peerId == p3)[0]
+    
+    check:
+       # regression on nim-libp2p fields
+      p3.addrs == @[MultiAddress.init("/ip4/127.0.0.1/tcp/3").tryGet()]
+      p3.protos == @["/vac/waku/lightpush/2.0.0", "/vac/waku/store/2.0.0-beta1"]
+      p3.agent == "gowaku"
+      p3.protoVersion == "protoVersion3"
+
+      # our extended fields
+      p3.connection == Connected
+      p3.disconnectTime == 0
+      p3.origin == Discv5
+
+  test "peers() returns all StoredInfo matching a specific protocol":
+    # When
+    let storePeers = peerStore.peers("/vac/waku/store/2.0.0")
+
+    # Then
+    check:
+      # Only p1 and p2 support that protocol
+      storePeers.len == 2
+      storePeers.anyIt(it.peerId == p1)
+      storePeers.anyIt(it.peerId == p2)
+
+    # When
+    let lpPeers = peerStore.peers("/vac/waku/lightpush/2.0.0")
+
+    # Then
+    check:
+      # Only p3 supports that protocol
+      lpPeers.len == 1
+      lpPeers.anyIt(it.peerId == p3)
+      lpPeers[0].protos == @["/vac/waku/lightpush/2.0.0", "/vac/waku/store/2.0.0-beta1"]
+
+  test "peers() returns all StoredInfo matching a given protocolMatcher":
+    # When
+    let pMatcherStorePeers = peerStore.peers(protocolMatcher("/vac/waku/store/2.0.0"))
+
+    # Then
+    check:
+      # peers: 1,2,3,5 match /vac/waku/store/2.0.0/xxx
+      pMatcherStorePeers.len == 4
+      pMatcherStorePeers.anyIt(it.peerId == p1)
+      pMatcherStorePeers.anyIt(it.peerId == p2)
+      pMatcherStorePeers.anyIt(it.peerId == p3)
+      pMatcherStorePeers.anyIt(it.peerId == p5)
+
+    check:
+      pMatcherStorePeers.filterIt(it.peerId == p1)[0].protos == @["/vac/waku/relay/2.0.0-beta1", "/vac/waku/store/2.0.0"]
+      pMatcherStorePeers.filterIt(it.peerId == p2)[0].protos == @["/vac/waku/relay/2.0.0", "/vac/waku/store/2.0.0"]
+      pMatcherStorePeers.filterIt(it.peerId == p3)[0].protos == @["/vac/waku/lightpush/2.0.0", "/vac/waku/store/2.0.0-beta1"]
+      pMatcherStorePeers.filterIt(it.peerId == p5)[0].protos == @["/vac/waku/swap/2.0.0", "/vac/waku/store/2.0.0-beta2"]
+
+    # When
+    let pMatcherSwapPeers = peerStore.peers(protocolMatcher("/vac/waku/swap/2.0.0"))
+
+    # Then
+    check:
+      pMatcherSwapPeers.len == 1
+      pMatcherSwapPeers.anyIt(it.peerId == p5)
+      pMatcherSwapPeers[0].protos == @["/vac/waku/swap/2.0.0", "/vac/waku/store/2.0.0-beta2"]
```

Comment:
The order should always be: "Given" -> "When" -> "Then".

All the checks should happen at the end of the test case under the _Then_ clause. This makes tests easier to maintain, and it will avoid chain reactions.

If there are any conditions _sine qua non_ that you need to assert within the _When_ block, then you must use `require`, which will fail the test case immediately. No `check` assertions must happen within a _When_ block.